File: CLion/Day3.cpp
#include<stdio.h>
int strlen(char * str) {
    int count=0;
    for (int i =0;str[i]!='\0';i++) {
        count++;
    }
    return count;
}
int strcmp(char *s1, char *s2) {
     if (strlen(s1) == strlen(s2)) {
         for (int i =0;i<strlen(s1);i++) {
            if (s1[i]!=s2[i]) {
                return 0;
            }
         }
         return 1;
     }
    return 0;
}
int main(void) {
    char paragraph[150] ="Bob hit a ball Bob hit a ball Bob hit a ball Bob hit a ball Bob hit a ball Bob hit a ball Bob hit a ball Bob hit a ball";
    char words[80][300];
    int wordcount=0;
    for (int i=0,j=0,index=0;i<sizeof(paragraph)-1;i++) {
        while (paragraph[i]!=' ' && i<sizeof(paragraph)-1) {
            words[index][j]=paragraph[i];
            i=i+1;
            j++;
        }
        words[index][i]='\0';
        index++;
        j=0;
        wordcount++;
    }
    char word[10]={"ball"};
    for (int i=0;i<wordcount;i++) {
        int j=0;
        while (words[i][j]!='\0') {
            printf("%c",words[i][j]);
            j++;
        }
        printf("\n");
    }
    printf("================================================\n");
    for (int i =0;i<wordcount;i++) {
        if (strcmp(words[i],word)!=0) {
            printf("Match found at the %d:\n",i);
        }
    }
}

File: Java/Day3.java
package Daywise;


public class Day3 {
    public static void main(String[] args){
        String  paragraph = "Bob hit an ball";
        char [][]words = new char[10][50];
        int wordcount=0;
        String word = "Bob";
        for(int i=0,j=0,index=0;i<paragraph.length();i++){
            while(paragraph.charAt(i)!=' '&& i<paragraph.length()){
               words[index][j]=paragraph.charAt(i);
               i=i+1;
               j=j+1;
            }
            words[index][j]='\0';
            index=index+1;
            wordcount=wordcount+1;
            j=0;

        }
        for(int i=0;i<wordcount;i++){
            int j =0;
            while(words[i][j]!='\0'){
                System.out.print(words[i][j]);
                j=j+1;
            }
            System.out.println();
        }
        for(int i =0;i<wordcount;i++){
            String currentWord = "";
            int j=0;
            while(words[i][j]!='\0') {
                currentWord =currentWord+ words[i][j];
                j++;
            }
            if (strcmp(currentWord, word) == 1) {
                System.out.println("Matched at " + i);
            }
        }
    }
   public static  int length(String str){
        int count=0;
        for(int i=0;str.charAt(i)!='\0';i++){
            count=count+1;
        }
        return count;
    }
    public static int strcmp(String str1,String str2){
        System.out.print(length(str1)+""+length(str2));
        if(length(str1)==length(str2)){

           for(int i=0;i<length(str1);i++){
               if(str1.charAt(i)!=str2.charAt(i)){
                   return 0;
               }
            }
           return 1;
        }
        return 0;
    }
}
